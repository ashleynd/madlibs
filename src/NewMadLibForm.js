import React, { useState } from "react";
// import Madlib from "./Madlib";

const NewMadLibForm = ({ addItem }) => {
    const INITIAL_STATE = {
        noun: '',
        noun2: '',
        adjective: '',
        color: ''
    }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addItem(formData.noun, formData.noun2, formData.adjective, formData.color);
        setFormData(INITIAL_STATE);
    }
    // const handleSubmit = (e) => {
    //     e.preventDefault();
    //     const {noun, noun2, adjective, color} = formData;
    //     alert(`The words inputted are ${noun} ${noun2} ${adjective} and ${color}`)
    //     setFormData(INITIAL_STATE);
    // }

    return (
        <form onSubmit={handleSubmit}>
            <p>
            <label htmlFor="noun"></label>
            <input 
            id="noun"
            type="text" 
            name="noun" 
            placeholder="Noun" 
            value={formData.noun} 
            onChange={handleChange} 
            required
            />
            </p>
            <p>
            <label htmlFor="noun2"></label>
            <input 
            id="noun2"
            type="text" 
            name="noun2" 
            placeholder="Noun 2" 
            value={formData.noun2} 
            onChange={handleChange} 
            required
            />
            </p>
            <p>
            <label htmlFor="adjective"></label>
            <input 
            id="adjective"
            type="text" 
            name="adjective" 
            placeholder="Adjective" 
            value={formData.adjective} 
            onChange={handleChange} 
            required
            />
            </p>
            <p>
            <label htmlFor="color"></label>
            <input 
            id="color"
            type="text" 
            name="color" 
            placeholder="Color" 
            value={formData.color} 
            onChange={handleChange} 
            required
            />
            </p>
            
            <button>Get Story</button>
        </form>
  
    )
}

export default NewMadLibForm;

