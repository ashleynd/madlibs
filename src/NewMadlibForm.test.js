import React from "react";
import { render } from "@testing-library/react";
import NewMadLibForm from "./NewMadLibForm";

it("renders without crashing", function() {
  render(<NewMadLibForm />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<NewMadLibForm />);
  expect(asFragment()).toMatchSnapshot();
});