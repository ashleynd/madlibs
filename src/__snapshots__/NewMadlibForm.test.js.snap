// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <p>
      <label
        for="noun"
      />
      <input
        id="noun"
        name="noun"
        placeholder="Noun"
        required=""
        type="text"
        value=""
      />
    </p>
    <p>
      <label
        for="noun2"
      />
      <input
        id="noun2"
        name="noun2"
        placeholder="Noun 2"
        required=""
        type="text"
        value=""
      />
    </p>
    <p>
      <label
        for="adjective"
      />
      <input
        id="adjective"
        name="adjective"
        placeholder="Adjective"
        required=""
        type="text"
        value=""
      />
    </p>
    <p>
      <label
        for="color"
      />
      <input
        id="color"
        name="color"
        placeholder="Color"
        required=""
        type="text"
        value=""
      />
    </p>
    <button>
      Get Story
    </button>
  </form>
</DocumentFragment>
`;
